src/backend/distributed/commands/begin.c

```diff
@@ -0,0 +1,49 @@
+/*-------------------------------------------------------------------------
+ *
+ * begin.c
+ *    Processing of the BEGIN command.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "c.h"
+
+#include "distributed/commands.h"
+#include "distributed/listutils.h"
+#include "distributed/transaction_management.h"
+#include "nodes/parsenodes.h"
+
+
+/*
+ * CitusProcessBegin handles TransactionStmt that has already been qualified
+ * as a BEGIN statement.
+ */
+void
+CitusProcessBegin(TransactionStmt *transactionStmt)
+{
+	DefElem *item = NULL;
+
+	/*
+	 * This loop is similar to the one in standard_ProcessUtility.
+	 *
+	 * While BEGIN can be quite frequent it will rarely have options set.
+	 */
+	foreach_ptr(item, transactionStmt->options)
+	{
+		A_Const *constant = (A_Const *) item->arg;
+
+		if (strcmp(item->defname, "transaction_read_only") == 0)
+		{
+			HasBeginXactReadOnly = true;
+			BeginXactReadOnly = (bool) intVal(&constant->val);
```

Comment:
Maybe I am missing something, but why not defGetBoolean()?

Reply:
these DefElems are somewhat weirdly structured for whatever reason 

rather than containing a proper Value like Integer or String like defGetBoolean expects, they contain A_Const, which is a parser artifact